### PR TITLE
Moving foam-specific content to `foam/` dir

### PR DESCRIPTION
**Pull request by cason**
Now we need a README.

This PR also enables the generation of proper markdown links from foam `[[links]]`. They are added by VS Code to the bottom of the files, see [docs/publishing/publish-to-github.md](https://github.com/cometbft/knowledge-base/blob/cason/foam-setup/foam/docs/publishing/publish-to-github.md).

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (thanethomson) [inline .vscode/extensions.json:1]: Should we be committing the `.vscode` folder? Won't this cause each of our local VS Code options to conflict every time we add and push changes?

Comment (cason) [reply .vscode/extensions.json:1]: I really don't know. Is this file shared by all extensions? I put it here because in the template it was present.

Comment (thanethomson) [reply .vscode/extensions.json:1]: I think this is something that we each need to set up locally, and we need a `.gitignore` file to ignore the `.vscode` directory. It stores local (personal) preferences/config, so I don't think it shouldn't be committed.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+  },
+  "files.defaultLanguage": "markdown",
+  "files.exclude": {
+    "**/node_modules": true
+  },
+  "files.watcherExclude": {
+    "**/node_modules": true
+  },
+  "foam.edit.linkReferenceDefinitions": "withExtensions",
```
Comment (cason) [inline .vscode/settings.json:20]: @thanethomson, how can I make this setting global? This and another pair are relevant here.

I will get read of other configs from vscode.

Comment (thanethomson) [reply .vscode/settings.json:20]: I think what you want is to configure your **User** settings. The `.vscode` folder here only configures **Workspace** settings. See https://code.visualstudio.com/docs/getstarted/settings

Comment (cason) [reply .vscode/settings.json:20]: Not really, I want this, which are essentially paths, only apply to this repository, not to override global/user configurations.

I don't know exactly what a Workspace means for vscode, so probably the best is to remove this dir, like you suggested.

Comment (thanethomson) [reply .vscode/settings.json:20]: Oh, you can still have a `.vscode` directory in your own local version of the repo. Please just don't commit it (i.e. add it to a `.gitignore` file).

**Comment by cason** [comment]
Dropping this PR in favor of #5.

**Pull request closed without merging**